src/applications/vaos/new-appointment/components/VAFacilityPage/VAFacilityPageV2.jsx

```diff
@@ -32,31 +38,156 @@ const pageKey = 'vaFacilityV2';
-    openFacilityPageV2(pageKey, uiSchema, initialSchema);
-  }, []);
-
-  if (childFacilitiesStatus === FETCH_STATUS.loading) {
-    return <LoadingIndicator message="Finding locations" />;
+  const history = useHistory();
+  const loadingEligibility = loadingEligibilityStatus === FETCH_STATUS.loading;
+  const loadingParents = parentFacilitiesStatus === FETCH_STATUS.loading;
+  const loadingFacilities = childFacilitiesStatus === FETCH_STATUS.loading;
+  const [showEligibilityModal, setShowEligibilityModal] = useState(false);
+
+  useEffect(
+    () => {
+      document.title = `${pageTitle} | Veterans Affairs`;
+      scrollAndFocus();
+      openFacilityPageV2(pageKey, uiSchema, initialSchema);
+    },
+    [openFacilityPageV2],
+  );
+
+  useEffect(
+    () => {
+      if (!loadingEligibility && canScheduleAtChosenFacility === false) {
+        setShowEligibilityModal(true);
+      }
+    },
+    [loadingEligibility, canScheduleAtChosenFacility],
+  );
+
+  const goBack = () => routeToPreviousAppointmentPage(history, pageKey);
+  const goForward = () => routeToNextAppointmentPage(history, pageKey);
+
+  const title = (
+    <h1 className="vads-u-font-size--h2">
+      Choose a VA location for your {typeOfCare} appointment
+    </h1>
+  );
+
+  if (
+    loadingParents ||
+    loadingFacilities ||
+    (singleValidVALocation && loadingEligibility)
+  ) {
+    return (
+      <div>
+        {title}
+        <LoadingIndicator message="Finding locations" />
+      </div>
+    );
+  }
+
+  if (noValidVAParentFacilities) {
+    return (
+      <div>
+        {title}
+        <NoVASystems />
+        <div className="vads-u-margin-top--2">
+          <FormButtons
+            onBack={goBack}
+            disabled
+            pageChangeInProgress={pageChangeInProgress}
+            loadingText="Page change in progress"
+          />
+        </div>
+      </div>
+    );
+  }
+
+  if (noValidVAFacilities) {
+    return (
+      <div>
+        {title}
+        <NoValidVAFacilities
+          formContext={{
+            siteId,
+            typeOfCare,
+            facilityDetailsStatus,
+            parentDetails,
+          }}
+        />
+        <div className="vads-u-margin-top--2">
+          <FormButtons
+            onBack={goBack}
+            disabled
+            pageChangeInProgress={pageChangeInProgress}
+            loadingText="Page change in progress"
+          />
+        </div>
+      </div>
+    );
+  }
+
+  if (singleValidVALocation && !canScheduleAtChosenFacility) {
+    return (
+      <div>
+        {title}
+        <SingleFacilityEligibilityCheckMessage
+          eligibility={eligibility}
+          facility={facility}
+        />
+        <div className="vads-u-margin-top--2">
+          <FormButtons
+            onBack={goBack}
+            disabled
+            pageChangeInProgress={pageChangeInProgress}
+            loadingText="Page change in progress"
+          />
+        </div>
+      </div>
+    );
+  }
```

Comment:
Displays messaging instead of radio buttons if user has 1 facility that accepts this type of care, but they are ineligible.